examples/multisig_plain/lib.rs

```diff
@@ -0,0 +1,837 @@
+        }
+
+        /// Remove all confirmation state associated with `owner`.
+        /// Also adjusts the `self.confirmation_count` variable.
+        fn clean_owner_confirmations(&mut self, owner: &AccountId) {
+            for (trans_id, _) in self.transactions.iter() {
+                if self.confirmations.remove(&(trans_id, *owner)).is_some() {
+                    mutate_map(&mut self.confirmation_count, &trans_id, |count| {
+                        *count += 1
+                    });
+                }
+            }
+        }
+
+        /// This removes all confirmation state associated with `transaction`.
+        fn clean_transaction_confirmations(&mut self, transaction: TransactionId) {
+            for owner in self.owners.iter() {
+                self.confirmations.remove(&(transaction, *owner));
+            }
+            self.confirmation_count.remove(&transaction);
+        }
+
+        /// Panic if transaction `trans_id` is not confirmed by at least
+        /// `self.requirement` owners.
+        fn ensure_confirmed(&self, trans_id: TransactionId) {
+            assert!(
+                self.confirmation_count
+                    .get(&trans_id)
+                    .expect(WRONG_TRANSACTION_ID)
+                    >= self.requirement.get()
+            );
+        }
+
+        /// Panic of the transaction `trans_id` does not exit.
+        fn ensure_transaction_exists(&self, trans_id: TransactionId) {
+            self.transactions.get(trans_id).expect(WRONG_TRANSACTION_ID);
+        }
+
+        /// Panic if the sender is no owner of the wallet.
+        fn ensure_caller_is_owner(&self) {
+            self.ensure_owner(&self.env().caller());
+        }
+
+        /// Panic if the sender is not this wallet.
+        fn ensure_from_wallet(&self) {
+            assert_eq!(self.env().caller(), self.env().account_id());
+        }
+
+        /// Panic if `owner` is not an owner,
+        fn ensure_owner(&self, owner: &AccountId) {
+            assert!(self.is_owner.contains_key(owner));
+        }
+
+        /// Panic if `owner` is an owner.
+        fn ensure_no_owner(&self, owner: &AccountId) {
+            assert!(!self.is_owner.contains_key(owner));
+        }
+
+        /// Panic if the number of `owners` under a `requirement` violates our
+        /// requirement invariant.
+        fn ensure_requirement_is_valid(&self, owners: u32, requirement: u32) {
+            assert!(0 < requirement && requirement <= owners && owners <= MAX_OWNERS);
+        }
+    }
+
+    /// Change a stored value by reinserting it.
+    fn mutate_map<K, V, F>(map: &mut storage::BTreeMap<K, V>, key: &K, f: F)
+    where
+        K: Eq + Ord + scale::Codec + Copy,
+        V: scale::Codec + Copy,
+        F: FnOnce(&mut V),
+    {
+        let mut count = *map
+            .get(key)
+            .expect("User is responsible for only supplying existing keys.");
+        f(&mut count);
+        map.insert(*key, count);
+    }
```

Comment:
Should we maybe have this in our `storage::BTreeMap` implementation? Is there a Rust `BTreeMap` equivalent

Reply:
I think we should. I was surprised that there was no `mutate_with` like with he `HashMap`.

Reply:
But on the other handside it has an entry API.

Reply:
I removed the function and switched to the entry API.